### ISSUE TITLE
Ideas: Custom Groups, Artist Categories, View Sorting

### ISSUE THREAD

**Issue by MrKuenning** [yfszzx/stable-diffusion-webui-inspiration#5]
First of all, thanks for this neat extension.  I have been looking for a visual way of remembering some of the artists and styles I like to use.
I have about 70 artists listed in my styles sheet with descriptions to try to remember, but as a visual person, this extension is a vast improvement.

I think this type of extension serves two purposes.

1. To discover or stumble upon new artists or ideas.  Hence the name inspiration.

2. To store or organize artists or styles that you want to commonly use for a quick call-back.

There are some features I would love to see added to help with these goals.

### Discovery

**Artist Tags**
On the stumbling-upon side of things, I would love to see Artist Categories or Tags.
Instead of pulling from 5000+ artists 24 at a time.  I would like to select comic-book, or renounce, and have it filter the list.

**Adjustable pool size and views**
I would also love to be able to set a higher number than 24 in the pool.

**Varying pool views**
Also maybe a list option with small thumbnails.  Especially with the flavors or mediums. I would like to see all of those at once and click on one for details on what it could look like.  Only seeing 24 makes me feel like it's withholding something.

**Pool sorting**
I would also very much like to be able to sort the pool alphabetically.  When I am searching for an artist by name, having it display a group of artists in random order can be frustrating.  A button to change from random to alphabetically would be nice.

### Organizing

**Custom Checkbox Groups**
I would like the option to create my own checkbox groups.  I would love to be able to sort and organize different styles or artists in ways to make it easier to visualize.

**Custom Source Items**
Similarly, I would like to be able to make custom Source dropdown items.  I could filter my top favorites, or maybe custom categories.

**Pool sorting**
As mentioned above the ability to sort the pool alphabetically is essential for custom-organized collections.

### Settings

There are also some settings would love to see, such as:
Choosing which checkbox groups are checked by default.
Choosing what prefix to put in front of an artist's name when sending to txt2img such as "art style as"

Again, thanks for the awesome extension. I am happy to provide better explanations or feedback.

**Comment by yfszzx** [comment]
Thanks for your advises. you can adjustable pool size and views in setting page.

**Comment by yfszzx** [comment]
Thanks for your advises. 

You can adjustable pool size and views in setting page.

Your suggestions are very pertinent, but some of them are technically difficult or take a lot of time. I will gradually add these suggestions to my project when the time is right

**Comment by MrKuenning** [comment]
So I discovered if you make new folders in the inspiration folder and restart SD it will automatically create new checkbox groups, which is fantastic!  

I was able to make one called embeddings and one called styles and create samples of  my embeddings and of my custom CSV styles.

![2022-10-25 10_26_34-Stable Diffusion](https://user-images.githubusercontent.com/16549268/197853133-cdad93c9-55c8-4324-8115-be9d6297640e.jpg)

This made me really excited about another feature.  The ability to set a custom complex prompt for each object/folder.

In the above screenshot, the selected style is called _Porcelain Sculpture_ which uses the following prompt:

`A Photo-real Delicate Ceramic Porcelain Sculpture of a {prompt}, In Front Of an Intricate Background By Victo Ngai And Takato Yamamoto, Micro Detail, Backlit Lighting, Subsurface Scattering, Translucent, Thin Porcelain, Octane Renderer, Colorful, Physically Based Rendering, Japanese Pottery, Trending On Cgsociety`

I was wondering if we could place a text file in each folder with a prompt and maybe also a description of the prompt.
We could use this extension as a visual prompt/style list.  

TLDR:
It would be awesome to store examples of complex prompts and apply those prompts from this extension.

**Comment by MrKuenning** [comment]
![2022-10-25 11_04_02-Stable Diffusion](https://user-images.githubusercontent.com/16549268/197860029-ba96e080-4794-43c3-80ea-20a2cc9a23ae.jpg)

Its awesome I also created a folder for my hypernetworks and my ckpt models.

**Comment by yfszzx** [comment]
> So I discovered if you make new folders in the inspiration folder and restart SD it will automatically create new checkbox groups, which is fantastic!
> 
> I was able to make one called embeddings and one called styles and create samples of my embeddings and of my custom CSV styles.
> 
> ![2022-10-25 10_26_34-Stable Diffusion](https://user-images.githubusercontent.com/16549268/197853133-cdad93c9-55c8-4324-8115-be9d6297640e.jpg)
> 
> This made me really excited about another feature. The ability to set a custom complex prompt for each object/folder.
> 
> In the above screenshot, the selected style is called _Porcelain Sculpture_ which uses the following prompt:
> 
> `A Photo-real Delicate Ceramic Porcelain Sculpture of a {prompt}, In Front Of an Intricate Background By Victo Ngai And Takato Yamamoto, Micro Detail, Backlit Lighting, Subsurface Scattering, Translucent, Thin Porcelain, Octane Renderer, Colorful, Physically Based Rendering, Japanese Pottery, Trending On Cgsociety`
> 
> I was wondering if we could place a text file in each folder with a prompt and maybe also a description of the prompt. We could use this extension as a visual prompt/style list.
> 
> TLDR: It would be awesome to store examples of complex prompts and apply those prompts from this extension.

“ I was wondering if we could place a text file in each folder with a prompt and maybe also a description of the prompt. “
good idea
